agent/stats/engine.go

```diff
@@ -163,77 +199,78 @@ func (engine *DockerStatsEngine) removeAll() {
 		}
 		delete(engine.tasksToContainers, task)
 	}
-}
-
-// listContainersAndStartEventHandler adds existing containers to the watch-list
-// and starts the docker event handler.
-func (engine *DockerStatsEngine) listContainersAndStartEventHandler() {
-	// List and add existing containers to the list of containers to watch.
-	err := engine.addExistingContainers()
-	if err != nil {
-		seelog.Warnf("Error listing existing containers, err: %v", err)
-		return
-	}
-}
-
-// addExistingContainers lists existing containers and adds them to the engine.
-func (engine *DockerStatsEngine) addExistingContainers() error {
-	listContainersResponse := engine.client.ListContainers(false, ecsengine.ListContainersTimeout)
-	if listContainersResponse.Error != nil {
-		return listContainersResponse.Error
-	}
 
-	for _, containerID := range listContainersResponse.DockerIDs {
-		engine.addContainer(containerID)
+	for task := range engine.tasksToHealthCheckContainers {
+		delete(engine.tasksToContainers, task)
 	}
-
-	return nil
 }
 
-// addContainer adds a container to the map of containers being watched.
-// It also starts the periodic usage data collection for the container.
-func (engine *DockerStatsEngine) addContainer(dockerID string) {
-	engine.lock.Lock()
-	defer engine.lock.Unlock()
-
+// addContainerUnsafe adds a container to the map of containers being watched.
```

Comment:
maybe i missed this. why are we getting rid of the locks here and using `*Unsafe`?

Reply:
No, the lock was just moved to line 138.

Reply:
oops not sure how i missed that. thanks.